plugin/main/src/kotlinx/benchmark/gradle/internal/KotlinxBenchmarkPluginInternalApi.kt

```diff
@@ -0,0 +1,16 @@
+package kotlinx.benchmark.gradle.internal
+
+/**
+ * Marks declarations that are **internal** to the Kotlinx Benchmark project.
+ *
+ * Classes or functions marked with this annotation have `public` visibility
+ * for historical or technical reasons. They are not intended for use by
+ * Kotlinx Benchmark users. They have no behaviour guarantees, and may lack clear
+ * semantics, documentation, and backward compatibility.
+ *
+ * Please remove references to internal API to supported stable references.
+ * If you still require access to these APIs, please raise an issue.
+ */
+@RequiresOptIn
```

Comment:
It might make sense to start with the `WARNING` level to make the migration process smother. WDYT?

Reply:
Good idea!